plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteInsertBatchImpl.java

```diff
@@ -159,54 +163,76 @@ StringBuilder prepareQueryForStatement(
             query.append(attributeName);
         }
         query.append(")\n\tVALUES "); //$NON-NLS-1$
+        
+        // Filtered attributes to remove columns that are ignored
+        DBSAttributeBase[] attributesf = new DBSAttributeBase[attributes.length - skipAttributes.size()];
+        Object[] attributeValuesf = new Object[attributeValues.length - (attributes.length * skipAttributes.size())];
+        DBDValueHandler[] handlersf = new DBDValueHandler[handlers.length - skipAttributes.size()];
+
+        if (skipAttributes.size() > 0) {
+            int j = -1;
+            for (int i = 0; i < attributes.length; i++) {
+                if (skipAttributes.contains(i)) {
+                    continue;
+                }
+                attributesf[++j] = attributes[i];
+            }
+            j = -1;
+            for (int i = 0; i < handlers.length; i++) {
+                if (skipAttributes.contains(i)) {
+                    continue;
+                }
+                handlersf[++j] = handlers[i];
+            }
+            int k = -1;
+            for (int i = 0; i < attributeValues.length; i++) {
+                j = i % attributes.length;
+                if (skipAttributes.contains(j)) {
+                    continue;
+                }
+                attributeValuesf[++k] = attributeValues[i];
+            }
+        } else {
+            attributesf = attributes;
+            handlersf = handlers;
+            attributeValuesf = attributeValues;
+        }
```

Comment:
Perhaps it may work, it's really bug-prone.

There's two ways to improve readability:
* Use `List<>` for storing filtered objects, that way you don't have to track the last indexes.
* Use `Arrays.stream(...)` and perform filtering in-place. Then use `.toArray` to convert stream into arrays.

And please use less-cryptic names for variables holding filtered objects. You can borrow original names and rename input parameters (just for this method) for something like `blabla` &rArr; `allBlaBla`

Reply:
Done!

Reply:
Done now! Applied the changes you suggested.